### PR TITLE
conferences_relation_table: Add conference_literature table

### PR DESCRIPTION
**Pull request by pazembrz**
INSPIR-2870

### CODE REVIEW

File: backend/inspirehep/records/api/mixins.py [hunk 1]
```diff
@@ -313,3 +317,43 @@ def is_bucket_uuid(test_str):
     @staticmethod
     def is_hash(test_str):
         return test_str and len(test_str) == 40
+
+
+class ConferencePaperAndProccedingsMixin:
```
Comment (michamos) [inline backend/inspirehep/records/api/mixins.py]:
```suggestion
class ConferencePaperAndProceedingsMixin:
```

File: backend/inspirehep/records/api/mixins.py [hunk 2]
```diff
@@ -313,3 +317,43 @@ def is_bucket_uuid(test_str):
...
+                    ConferenceLiterature(
+                        conference_uuid=conference.id,
+                        literature_uuid=self.id,
+                        relationship_type=ConferenceToLiteratureRelationshipType(
+                            document_type
+                        ),
+                    )
+                )
+
+    def update_conference_paper_and_proccedings(self):
```
Comment (michamos) [inline backend/inspirehep/records/api/mixins.py]:
```suggestion
    def update_conference_paper_and_proceedings(self):
```

File: backend/inspirehep/records/api/mixins.py [hunk 3]
```diff
@@ -313,3 +317,43 @@ def is_bucket_uuid(test_str):
...
+        self.clean_conference_literature_relation()
+        document_type = self.get("document_type")[0]
```
Comment (michamos) [inline backend/inspirehep/records/api/mixins.py]: this is wrong, it doesn't have to be the first in the list, you should check if any is `proceedings` or `conference paper`.

**Comment by michamos** [comment]
and all the reindexing of dependent records, but maybe that's a different story.

**Comment by pazembrz** [comment]
There is no indexing/reindexing stuff as there are no serializers yet. 
There is nothing for migration as this is separate task: https://its.cern.ch/jira/browse/INSPIR-2867

**Comment by pazembrz** [comment]
Actually you are right about migration. I should add flags to disable this during migration.

File: backend/inspirehep/migrator/tasks.py [hunk 4]
```diff
@@ -275,6 +279,11 @@ def process_references_in_records(uuids):
     return uuids
 
 
+@shared_task(ignore_results=False, queue="migrator", acks_late=True)
+def process_conferences_papers(uuids):
+    return process_record_linked_to_conferences(uuids)
```
Comment (michamos) [inline backend/inspirehep/migrator/tasks.py]: I think it's better to generalize the `recalculate_citations` into `recalculate_records_relations` in which you do both.

File: backend/inspirehep/migrator/tasks.py [hunk 5]
```diff
@@ -315,18 +324,27 @@ def insert_into_mirror(raw_records):
 
 
 def migrate_and_insert_record(
-    raw_record, disable_orcid_push=False, disable_citation_update=False
+    raw_record,
+    disable_orcid_push=False,
+    disable_citation_update=False,
+    disable_conference_update=False,
```
Comment (michamos) [inline backend/inspirehep/migrator/tasks.py]: have only one flag `disable_relations_update`?

File: backend/tests/integration/records/api/test_api_mixins.py [hunk 6]
```diff
@@ -292,3 +285,189 @@ def test_regression_add_original_url_if_it_is_external(
...
+    }
+    rec = create_record("lit", rec_data)
+
+    rec.hard_delete()
+
+    assert len(conference.model.conference_documents) == 0
+
+
+@mock.patch.object(LiteratureRecord, "update_conference_paper_and_proccedings")
+def test_dsiable_donference_update_switch_works(
```
Comment (ammirate) [inline backend/tests/integration/records/api/test_api_mixins.py]: `test_disable_conference_update_feature_flag_disabled`

File: backend/inspirehep/records/api/mixins.py [hunk 7]
```diff
@@ -313,3 +317,50 @@ def is_bucket_uuid(test_str):
...
+    def hard_delete(self):
+        self.clean_conference_literature_relation()
+        super().hard_delete()
+
+    def update(self, data, disable_conference_update=False, *args, **kwargs):
+        super().update(data, *args, **kwargs)
+        if not disable_conference_update:
+            self.update_conference_paper_and_proccedings()
+        else:
+            LOGGER.info(
```
[the remark this answers is not in the document]
Comment (pazembrz) [reply backend/inspirehep/records/api/mixins.py]: We have same level for citations, so I will leave it this way for consistency.

File: backend/inspirehep/migrator/tasks.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (pazembrz) [reply backend/inspirehep/migrator/tasks.py]: Now i merged citations with conference relations.

File: backend/inspirehep/records/tasks.py [hunk 8]
```diff
@@ -36,6 +36,30 @@ def recalculate_record_citations(uuids):
     return uuids
 
 
+def process_record_linked_to_conferences(uuids):
```
[the remark this answers is not in the document]
Comment (pazembrz) [reply backend/inspirehep/records/tasks.py]: up

**Pull request merged**